src/main/java/com/igot/cb/consentacknowledge/ConsentAcknowledgeServiceImpl.java

```diff
@@ -0,0 +1,155 @@
+    private final CassandraOperation cassandraOperation;
+    private final ObjectMapper objectMapper;
+
+    public ConsentAcknowledgeServiceImpl(AccessTokenValidator accessTokenValidator, CassandraOperation cassandraOperation, ObjectMapper objectMapper) {
+        this.cassandraOperation = cassandraOperation;
+        this.accessTokenValidator = accessTokenValidator;
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * Method to acknowledge declaration
+     * Required fields in request body : contentId, consentId
+     * Optional fields in request body : additionalData
+     */
+    @Override
+    public ApiResponse acknowledgeDeclaration(Map<String, Object> requestDataBody, String authToken) {
+        ApiResponse response = ProjectUtil.createDefaultResponse("api.consent.acknowledge.create");
+        Map<String, Object> requestDataMap = (Map<String, Object>) requestDataBody.get(Constants.REQUEST);
+        String userId = accessTokenValidator.fetchUserIdFromAccessToken(authToken, response);
+        if (StringUtils.isEmpty(userId)) {
+            return response;
+        }
+        String errMsg = validateAcknowledgeDeclarationRequest(requestDataMap);
+        if (!StringUtils.isEmpty(errMsg)) {
+            ProjectUtil.errorResponse(response, errMsg, HttpStatus.BAD_REQUEST);
+            return response;
+        }
+        String contentId = (String) requestDataMap.get(Constants.CONTENT_ID);
+        String consentId = (String) requestDataMap.get(Constants.CONSENT_ID);
+        Map<String, Object> additionalData = (Map<String, Object>) requestDataMap.get(Constants.ADDITIONAL_ATTRIBUTES);
+        String additionalDataStr = null;
+        try {
+            additionalDataStr = objectMapper.writeValueAsString(additionalData);
+        } catch (JsonProcessingException e) {
+            logger.error("Error while converting additionalData to string", e);
+            ProjectUtil.errorResponse(response,"Failed to Parse the additional attributes", HttpStatus.BAD_REQUEST);
+            return response;
+        }
+        Map<String, Object> compositeKeyMap = new HashMap<>();
+        compositeKeyMap.put(Constants.USER_ID, userId);
+        compositeKeyMap.put(Constants.CONSENT_ID, consentId);
+        Map<String, Object> declarationDataMap = new HashMap<>();
+        ZoneId zoneId = ZoneId.of("Asia/Kolkata");
+        ZonedDateTime submittedAt = ZonedDateTime.ofInstant(Instant.now(), zoneId);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
+        declarationDataMap.put(Constants.SUBMITTED_BY, userId);
+        declarationDataMap.put(Constants.ADDITIONAL_ATTRIBUTES, additionalDataStr);
+        declarationDataMap.put(Constants.SUBMITTED_AT, submittedAt.format(formatter));
+        try {
+            cassandraOperation.insertRecord(Constants.KEYSPACE_SUNBIRD, Constants.TABLE_DECLARATION_ACKNOWLEDGMENT, Constants.CONTENT_ID, contentId, compositeKeyMap, declarationDataMap);
```

Comment:
when there is an exception -- insertRecord doesn't re-throw.
either you need to re-throw the exception or validate the return response.